aws/resource_aws_msk_cluster.go

```diff
@@ -106,6 +110,19 @@ func resourceAwsMskCluster() *schema.Resource {
 								},
 							},
 						},
+						"sasl": {
+							Type:     schema.TypeList,
+							Optional: true,
```

Comment:
Since there can only be one possible list configuration for `sasl` , we should also mark this with `MaxItems: 1` similar to the `tls` field above..in addition, I'm not see the parent field `client_authentication` can be updated as it's marked with `ForceNew`, so let's also mark this field with `ForceNew: true` and its' child field, `scram` below should also include this `ForceNew` setting